src/Microsoft.AspNet.Mvc.Core/ApplicationModels/DefaultControllerModelBuilder.cs

```diff
@@ -126,6 +130,12 @@ protected virtual ControllerModel CreateControllerModel([NotNull] TypeInfo typeI
             AddRange(controllerModel.Filters, attributes.OfType<IFilter>());
             AddRange(controllerModel.RouteConstraints, attributes.OfType<IRouteConstraintProvider>());
 
+            var policy = AuthorizationPolicy.Combine(_authorizationOptions, attributes.OfType<AuthorizeAttribute>());
+            if (policy != null)
+            {
+                controllerModel.Filters.Add(new AuthorizeFilter(policy));
```

Comment:
I don't understand the comment here. `IFilterFactory` is only useful when when a filter needs indirection (replaceable) or needs access to request services. 

If you made this `IFilterFactory` you'd still need code here because `AuthorizeAttribute` cannot implement `IFilterFactory` itself.